portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +32,50 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    String commentLimitString = request.getParameter("limit");
+    int commentLimit = Integer.parseInt(commentLimitString);
+    if (commentLimit <= 0) commentLimit = 5;
+    for (Entity entity : results.asIterable()) {
+      String text = (String) entity.getProperty("text");
```

Comment:
I think it would be nice to extract this conversion from `Entity` -> `Comment` into it's own method. The details of the conversion aren't really important to the logic of this method and then you can also re-use it later if you need it.

Reply:
Done.